### PR TITLE
fix druid connector with tests

### PR DESCRIPTION
**Pull request by cosmic-chichu**
- decouple method to increase testability
-  remove extraneous subquery

### CODE REVIEW

File: numalogic/connectors/druid.py [hunk 1]
```diff
@@ -15,6 +15,67 @@
...
+    Returns: a dict of key value pairs
+
+    """
+    filter_pairs = {}
+    for k, v in zip(filter_keys, filter_values):
+        filter_pairs[k] = v
+    return filter_pairs
+
+
+def build_params(aggregations, datasource, dimensions, filter_pairs, granularity, hours):
```
Comment (s0nicboOm) [inline numalogic/connectors/druid.py:34]: Can we describe typing for variables please.
Everyone is not druid expert :)

File: numalogic/connectors/druid.py [hunk 2]
```diff
@@ -15,6 +15,67 @@
...
+        dimensions: The dimensions to group by
+        filter_pairs: Indicates which rows of
+          data to include in the query
+        granularity: Time bucket to aggregate data by hour, day, minute, etc.,
+        hours: Hours from now to skip training.
+
+    Returns: a dict of parameters
+
+    """
+    _start_time = time.time()
```
Comment (s0nicboOm) [inline numalogic/connectors/druid.py:50]: I think we are having this for Logging purpose.
We are not finding end_time and printing it.

**Review by s0nicboOm — COMMENTED** [review]
Minor comments

File: numalogic/connectors/druid.py (hunk 1 above)
Comment (ab93) [reply numalogic/connectors/druid.py:34]: +1

File: tests/connectors/test_druid.py [hunk 3]
```diff
@@ -60,6 +63,21 @@ def test_fetch_data(self):
...
+            "filter": [Filter(type="selector", dimension="ciStatus", value="false")],
+            "granularity": "all",
+            "aggregations": "",
+            "intervals": "",
+        }
+
+        filter_pairs = make_filter_pairs(["ciStatus"], ["false"])
+        actual = build_params("", "foo", ["bar"], filter_pairs, "all", float(24))
+        actual["intervals"] = ""
+        DeepDiff(expected, actual).get("values_changed", {})
```
Comment (ab93) [inline tests/connectors/test_druid.py:79]: Are we doing an assert here? If `values_changed` is non zero, will this test fail?

**Review by ab93 — CHANGES_REQUESTED** [review]
Some comments

File: numalogic/connectors/druid.py [hunk 4]
```diff
@@ -15,6 +15,67 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+def make_filter_pairs(filter_keys, filter_values):
```
Comment (ab93) [inline numalogic/connectors/druid.py:18]: Let's add typing here too please

File: numalogic/connectors/druid.py (hunk 1 above)
Comment (cosmic-chichu) [reply numalogic/connectors/druid.py:34]: cool adding type hints

File: numalogic/connectors/druid.py [hunk 5]
```diff
@@ -15,6 +15,67 @@
...
+        filter_keys: keys
+        filter_values: values.
+
+    Returns: a dict of key value pairs
+
+    """
+    filter_pairs = {}
+    for k, v in zip(filter_keys, filter_values):
+        filter_pairs[k] = v
+    return filter_pairs
```
Comment (ab93) [inline numalogic/connectors/druid.py:31]:
```suggestion
    return dict(zip(filter_keys, filter_values))
```

File: numalogic/connectors/druid.py [hunk 6]
```diff
@@ -15,6 +15,70 @@
 _LOGGER = logging.getLogger(__name__)
 
 
+# TODO: pass dictionary of keys and values as dict
+def make_filter_pairs(filter_keys: list[str], filter_values: list[str]) -> dict:
```
Comment (ab93) [inline numalogic/connectors/druid.py:19]:
```suggestion
def make_filter_pairs(filter_keys: list[str], filter_values: list[str]) -> dict[str, str]:
```

File: numalogic/connectors/druid.py [hunk 7]
```diff
@@ -15,6 +15,70 @@
...
+
+
+def build_params(
+    aggregations: list[str],
+    datasource: str,
+    dimensions: list[str],
+    filter_pairs: dict,
+    granularity: str,
+    hours: float,
+) -> object:
```
Comment (ab93) [inline numalogic/connectors/druid.py:39]:
```suggestion
) -> dict:
```

File: numalogic/connectors/druid.py [hunk 8]
```diff
@@ -90,6 +130,12 @@ def fetch(
             )
             df.columns = df.columns.map("{0[1]}".format)
             df.reset_index(inplace=True)
+
+        _end_time = time.perf_counter() - _start_time
+        _LOGGER.debug(
+            "Druid query latency: %s",
+            f"{_end_time:.6f}s for the calculation",
```
Comment (ab93) [inline numalogic/connectors/druid.py:137]: Can you convert this to a C style formatting?

File: tests/connectors/test_druid.py [hunk 9]
```diff
@@ -60,6 +63,22 @@ def test_fetch_data(self):
...
+            "granularity": "all",
+            "aggregations": "",
+            "intervals": "",
+        }
+
+        filter_pairs = make_filter_pairs(["ciStatus"], ["false"])
+        actual = build_params("", "foo", ["bar"], filter_pairs, "all", float(24))
+        actual["intervals"] = ""
+        diff = DeepDiff(expected, actual).get("values_changed", {})
+        self.assertEqual({}, diff)
```
Comment (ab93) [inline tests/connectors/test_druid.py:80]:
```suggestion
        self.assertDictEqual({}, diff)
```

**Review by ab93 — COMMENTED** [review]
Some minor comments

**Review by ab93 — APPROVED** [review]

**Review by nkoppisetty — APPROVED** [review]

**Pull request merged**